src/renderer/extensions/vueNodes/widgets/components/WidgetSelectButton.test.ts

```diff
@@ -0,0 +1,426 @@
+      const widget = createMockWidget('valid', { values: options })
+      const wrapper = mountComponent(widget, 'valid')
+
+      const buttons = wrapper.findAll('button')
+      expect(buttons).toHaveLength(4)
+      expect(buttons[0].classes()).toContain('bg-white')
+    })
+
+    it('handles very long option text', () => {
+      const longText =
+        'This is a very long option text that might cause layout issues if not handled properly'
+      const options = ['short', longText, 'normal']
+      const widget = createMockWidget('short', { values: options })
+      const wrapper = mountComponent(widget, 'short')
+
+      const buttons = wrapper.findAll('button')
+      expect(buttons[1].text()).toBe(longText)
+    })
+
+    it('handles large number of options', () => {
+      const options = Array.from({ length: 20 }, (_, i) => `option${i + 1}`)
+      const widget = createMockWidget('option5', { values: options })
+      const wrapper = mountComponent(widget, 'option5')
+
+      const buttons = wrapper.findAll('button')
+      expect(buttons).toHaveLength(20)
+      expect(buttons[4].classes()).toContain('bg-white') // option5 is at index 4
+    })
+
+    it('handles duplicate options', () => {
+      const options = ['duplicate', 'unique', 'duplicate', 'unique']
+      const widget = createMockWidget('duplicate', { values: options })
+      const wrapper = mountComponent(widget, 'duplicate')
+
+      const buttons = wrapper.findAll('button')
+      expect(buttons).toHaveLength(4)
+      // Both 'duplicate' buttons should be highlighted (due to value matching)
+      expect(buttons[0].classes()).toContain('bg-white')
+      expect(buttons[2].classes()).toContain('bg-white')
+    })
+  })
+
+  describe('Styling and Layout', () => {
+    it('applies proper button styling', () => {
+      const options = ['option1', 'option2']
+      const widget = createMockWidget('option1', { values: options })
+      const wrapper = mountComponent(widget, 'option1')
+
+      const buttons = wrapper.findAll('button')
+      buttons.forEach((button) => {
+        expect(button.classes()).toContain('flex-1')
+        expect(button.classes()).toContain('h-6')
+        expect(button.classes()).toContain('px-5')
+        expect(button.classes()).toContain('rounded')
+        expect(button.classes()).toContain('text-center')
+        expect(button.classes()).toContain('text-xs')
+      })
+    })
+
+    it('applies container styling', () => {
+      const options = ['option1', 'option2']
+      const widget = createMockWidget('option1', { values: options })
+      const wrapper = mountComponent(widget, 'option1')
+
+      const container = wrapper.find('div').element
+      expect(container.className).toContain('p-1')
+      expect(container.className).toContain('inline-flex')
+      expect(container.className).toContain('justify-center')
+      expect(container.className).toContain('items-center')
+      expect(container.className).toContain('gap-1')
```

Comment:
This feels very change-detectory, I'd remove tests that don't relate to the logic in the component.